File: Python/selectionSort.py
from assets.helper import print_stat


def findSmallest(list):
    smallest = list[0]
    smallest_index = 0
    for i in range(1, len(list)):
        if (list[i]) < smallest:
            smallest = list[i]
            smallest_index = i
    return smallest_index


def selectionSort(list):
    steps = 0
    newArr = []
    for item in range(len(list)):
        small = findSmallest(list)
        newArr.append(list.pop(small))
        steps += 1
    print_stat(len(newArr), steps)
    return newArr


res = selectionSort([5, 3, 6, 2, 10, 4, 8, 9, 22, 19])
print(res)

File: Java/SelectionSort.java
package Java;

import java.util.Arrays;

import Java.assets.Algorithm;
import Java.assets.Helper;

class SelectionSort implements Algorithm {
    private int steps = 0;
    private int[] arr;

    @Override
    public int[] getStats() {
        return new int[] { this.arr.length, this.steps };
    }

    public int[] selectionSort(int[] arr) {

        this.arr = arr;
        for (int i = 0; i < arr.length - 1; i++) {
            int index = i;
            for (int j = i + 1; j < arr.length; j++) {
                if (arr[j] < arr[index]) {
                    index = j;// searching for lowest index
                }
                ++steps;
            }
            int smallerNumber = arr[index];
            arr[index] = arr[i];
            arr[i] = smallerNumber;
        }
        return this.arr;
    }

    public static void main(String[] args) {
        SelectionSort alg = new SelectionSort();
        int[] arr = { 5, 3, 6, 2, 10, 4, 8, 9, 22, 19 };
        int[] res = alg.selectionSort(arr);
        Helper.print(alg.getStats());
        System.out.println(Arrays.toString(res));
    }
}